tests/test_surface/test_regular_surface_resample.py

```diff
@@ -71,15 +71,45 @@ def test_resample_small():
 
     xsx = xs1.copy()
     xsx.resample(xs2)
-    assert xsx.values.mean() == 888.0
+    assert list(xsx.values.data.flatten()) == [
```

Comment:
Added full assert on this, on master this will generally fail, but it will fail for different reasons as it relies on undefined behavior. The changes in this PR will make this consistent.